### PR TITLE
[JUJU-4187] Issue & PR templates

### PR DESCRIPTION
**Pull request by cderici**
## Description

This updates the pull request template and replaces the issue template (that's using old style markdown) with new issue templates separately for bug reports and feature requests using the new Github forms.

Adds a type of change to the PR template for maintenance work.

## QA steps

No complicated QA is needed, maybe wording could be checked. Also in the `Files Changed` section, there's a `View File` button in the top right `...` menu for individual files, which is useful to see what the Gİthub form will look like.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+labels: [wishlisted]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for suggesting this. Please provide info below so we can 
+        quickly add it to the terraform juju provider.
+
+  - type: textarea
+    id: Description
```
Comment (hmlanigan) [inline .github/ISSUE_TEMPLATE/FeatureRequest.yml:12]: should be required.

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    id: info
+    attributes:
+      label: "Reproduce / Test"
+      description: "Please add the commands to run to use the feature."
+      placeholder: ex. terraform init && terraform apply
+      render: shell
```
Comment (hmlanigan) [inline .github/ISSUE_TEMPLATE/FeatureRequest.yml:56]: Not sure this is relevant to the terraform provider

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  - type: dropdown
+    id: Urgency
+    attributes:
+      label: Urgency
+      description: How urgent is this issue for you?
+      options:
+        - Casually reporting
+        - Annoying bug in our test suite
+        - Blocker for our release
+        - Client presentation, help now please!
```
[the remark this answers is not in the document]
Comment (cderici) [reply .github/ISSUE_TEMPLATE/BugReport.yml:26]: sure

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      required: true
+
+  - type: input
+    id: Terraform-Version
+    attributes:
+      label: Terraform version
+      description: What version of Terraform are we running against? `terraform -v`
+      placeholder: ex. v1.5.2
+    validations:
+      required: true
```
[the remark this answers is not in the document]
Comment (cderici) [reply .github/ISSUE_TEMPLATE/FeatureRequest.yml:38]: agreed

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for suggesting this. Please provide info below so we can 
+        quickly add it to the terraform juju provider.
+
+  - type: textarea
+    id: Description
+    attributes:
+      label: Description
```
[the remark this answers is not in the document]
Comment (cderici) [reply .github/ISSUE_TEMPLATE/FeatureRequest.yml:14]: I don't understand what you mean

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+  - type: textarea
+    id: info
+    attributes:
+      label: "Code I'd Like to Run"
+      description: "Please add/copy-paste your Terraform plans (.tf) to
+      demonstrate the new feature you'd like to use."
+      render: terraform
+    validations:
+      required: true
```
[the remark this answers is not in the document]
Comment (cderici) [reply .github/ISSUE_TEMPLATE/FeatureRequest.yml:48]: not sure what's exactly on your mind, but changing it towards your suggestion 👍

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (hmlanigan) [reply .github/ISSUE_TEMPLATE/FeatureRequest.yml:14]: We have a place to put requirements etc, therefore we don't need another Description. Thus suggested changing to Title? Or may not needed.

**Review by hmlanigan — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/FeatureRequest.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (cderici) [reply .github/ISSUE_TEMPLATE/FeatureRequest.yml:14]: Looks like I forgot about this, I agree that field is redundant.

**Pull request merged**